src/transport/receiver.ts

```diff
@@ -45,44 +49,42 @@ export class HTTPReceiver {
    * @param {Object|JSON} body The body of the HTTP request
    * @return {CloudEvent} A new {CloudEvent} instance
    */
-  accept(headers: {}, body: {}) : CloudEvent {
+  accept(headers: Headers, body: string | object | CloudEventV1 | CloudEventV03): CloudEvent {
     const mode: string = getMode(headers);
     const version = getVersion(mode, headers, body);
     switch (version) {
-      case SPEC_V1:
+      case Version.V1:
         return this.receivers.v1[mode].parse(body, headers);
-      case SPEC_V03:
+      case Version.V03:
         return this.receivers.v03[mode].parse(body, headers);
       default:
-        console.error(`Unknown spec version ${version}. Default to ${SPEC_V1}`);
+        console.error(`Unknown spec version ${version}. Default to ${Version.V1}`);
         return this.receivers.v1[mode].parse(body, headers);
     }
   }
 }
 
-function getMode(headers: { [key: string]: string }) {
-  const contentType = headers[HEADER_CONTENT_TYPE];
-  if (contentType && contentType.startsWith(MIME_CE)) {
-    return STRUCTURED;
-  }
-  if (headers[BINARY_HEADERS_1.ID]) {
-    return BINARY;
+function getMode(headers: Headers): string {
+  const contentType = headers["content-type"];
+  if (contentType && contentType.startsWith("application/cloudevents")) {
+    return "structured";
   }
+  if (headers["ce-id"]) { return "binary"; }
   throw new ValidationError("no cloud event detected");
 }
 
-function getVersion(mode: string, headers: { [key: string]: string }, body: string | { specversion?: string }) {
-  if (mode === BINARY) {
+function getVersion(mode: string, headers: Headers, body: string | object | CloudEventV03 | CloudEventV1) {
+  if (mode === "binary") {
     // Check the headers for the version
-    const versionHeader = headers[DEFAULT_SPEC_VERSION_HEADER];
+    const versionHeader = headers["ce-specversion"];
```

Comment:
I think keeping the constants (although probably renamed) would be useful rather than raw strings.

Reply:
Agree - I just hadn't normalized my usage of the `constants.ts` file yet. That's in the TODO list in the PR description. I didn't like the way I was having to import them, so I punted and used strings until I had time to go back and figure out the best way to surface all of them. I personally don't think there should be a single `constants.ts` file for all constants in the project. Rather, I think they should be grouped accordingly - e.g. all of the header constants in `transport/http` somewhere.